hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hss/signschedule/SignScheduleTranslator.java

```diff
@@ -16,32 +16,171 @@
+    public static final Function SIGN_SCHEDULE_PROXY = new Function("signSchedule()", ReturnTypes.INT_64);
+    public static final Function AUTHORIZE_SCHEDULE = new Function("authorizeSchedule(address)", ReturnTypes.INT_64);
+
+    @Inject
+    public SignScheduleTranslator() {
+        // Dagger2
+    }
 
     @Override
     public boolean matches(@NonNull final HssCallAttempt attempt) {
         requireNonNull(attempt);
-        return attempt.isSelector(SIGN_SCHEDULE);
+        final var signScheduleEnabled =
+                attempt.configuration().getConfigData(ContractsConfig.class).systemContractSignScheduleEnabled();
+        final var authorizeScheduleEnabled =
+                attempt.configuration().getConfigData(ContractsConfig.class).systemContractAuthorizeScheduleEnabled();
+        return attempt.isSelectorIfConfigEnabled(signScheduleEnabled, SIGN_SCHEDULE_PROXY)
+                || attempt.isSelectorIfConfigEnabled(authorizeScheduleEnabled, AUTHORIZE_SCHEDULE);
     }
 
     @Override
     public Call callFrom(@NonNull HssCallAttempt attempt) {
-        return new SignScheduleCall(attempt);
+        final var body = bodyFor(scheduleIdFor(attempt));
+        return new DispatchForResponseCodeHssCall(
+                attempt, body, SignScheduleTranslator::gasRequirement, keySetFor(attempt));
+    }
+
+    /**
+     * Calculates the gas requirement for a {@code signSchedule()} call.
+     *
+     * @param body the transaction body
+     * @param systemContractGasCalculator the gas calculator
+     * @param enhancement the enhancement
+     * @param payerId the payer ID
+     * @return the gas requirement
+     */
+    public static long gasRequirement(
+            @NonNull final TransactionBody body,
+            @NonNull final SystemContractGasCalculator systemContractGasCalculator,
+            @NonNull final HederaWorldUpdater.Enhancement enhancement,
+            @NonNull final AccountID payerId) {
+        return systemContractGasCalculator.gasRequirement(body, DispatchType.SCHEDULE_SIGN, payerId);
+    }
+
+    /**
+     * Creates a transaction body for a {@code signSchedule()} or call.
+     *
+     * @param scheduleID the schedule ID
+     * @return the transaction body
+     */
+    private TransactionBody bodyFor(@NonNull ScheduleID scheduleID) {
+        requireNonNull(scheduleID);
+        return TransactionBody.newBuilder()
+                .scheduleSign(ScheduleSignTransactionBody.newBuilder()
+                        .scheduleID(scheduleID)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Extracts the schedule ID from a {@code authorizeSchedule()} call or return the redirect schedule ID
+     * if the call via the proxy contract
+     *
+     * @param attempt the call attempt
+     * @return the schedule ID
+     */
+    private ScheduleID scheduleIdFor(@NonNull HssCallAttempt attempt) {
+        requireNonNull(attempt);
+        if (attempt.isSelector(SIGN_SCHEDULE_PROXY)) {
```

Comment:
again why `SIGN_SCHEDULE_PROXY` here instead of `SIGN_SCHEDULE`?  (Or _additionally_ `SIGN_SCHEDULE`)

Reply:
see above.